### PR TITLE
[feature/CORE-QA-5] Sticky note automation

### LINKED ISSUE #20: [CORE-QA-5] Sticky note automation

### PR DESCRIPTION
**Pull request by jonathan016**
Fixes #20

### CODE REVIEW

File: src/main/java/com/future/function/qa/api/sticky_note/StickyNoteAPI.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import io.restassured.specification.RequestSpecification;
+import net.thucydides.core.annotations.Step;
+
+public class StickyNoteAPI extends BaseAPI {
+
+  @Step
+  public Response createStickyNote(StickyNoteWebRequest request, Cookie cookie) {
+
+    if (cookie == null) {
+      return createStickyNoteWithoutCookie(request);
```
Comment (oliverrsebastian) [inline src/main/java/com/future/function/qa/api/sticky_note/StickyNoteAPI.java:19]: maybe if future tests is going to be like this, can be moved to a new helper class

File: src/main/java/com/future/function/qa/data/BaseData.java [hunk 2]
```diff
@@ -1,27 +1,73 @@
...
+
   protected int responseCode;
 
-  public static BaseResponse asBaseResponse(Response response) {
+  public <T> DataResponse<T> asDataResponse(Response response, TypeReference<DataResponse<T>> typeReference) {
+
+    try {
+      return mapper.readValue(response.print(), typeReference);
+    } catch (Exception e) {
+      return new DataResponse<>();
```
Comment (oliverrsebastian) [inline src/main/java/com/future/function/qa/data/BaseData.java:36]: can this actually throw exception?

**Review by oliverrsebastian — CHANGES_REQUESTED** [review]

File: src/main/java/com/future/function/qa/api/sticky_note/StickyNoteAPI.java (hunk 1 above)
Comment (jonathan016) [reply src/main/java/com/future/function/qa/api/sticky_note/StickyNoteAPI.java:19]: Will change in future PR, maybe in CORE-QA-3

File: src/main/java/com/future/function/qa/data/BaseData.java (hunk 2 above)
Comment (jonathan016) [reply src/main/java/com/future/function/qa/data/BaseData.java:36]: Yes, so the idea was to have only 1 setter method for retrieved response (`BaseResponse`, `ErrorResponse`, `DataResponse`, or `PagingResponse`), which would set all those types of responses, should the retrieved response could be parsed. If not, then default value will be used (in this case: empty constructor)

**Review by oliverrsebastian — APPROVED** [review]

**Pull request merged**